x-pack/plugins/apm/public/components/shared/EnvironmentFilter/index.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiSelect, EuiFormLabel } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import styled from 'styled-components';
+import { useFetcher } from '../../../hooks/useFetcher';
+import { loadServiceEnvironments } from '../../../services/rest/apm/services';
+import { useLocation } from '../../../hooks/useLocation';
+import { useUiFilters } from '../../../hooks/useUiFilters';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { history } from '../../../utils/history';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+import {
+  ENVIRONMENT_ALL,
+  ENVIRONMENT_NOT_DEFINED
+} from '../../../../common/environment_filter_values';
+
+function updateEnvironmentUrl(
+  location: ReturnType<typeof useLocation>,
+  environment?: string
+) {
+  const nextEnvironmentQueryParam =
+    environment !== ENVIRONMENT_ALL ? environment : undefined;
+  history.push({
+    ...location,
+    search: fromQuery({
+      ...toQuery(location.search),
+      environment: nextEnvironmentQueryParam
+    })
+  });
+}
+
+function getOptions(environments: string[]) {
+  const SHOW_ALL_OPTION = {
```

Comment:
nit:can we call this `ENVIRONMENT_ALL_OPTION` or `ALL_OPTION` so it aligns with the name of the value (`ENVIRONMENT_ALL`)?